File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/trending/model/TrendingTimeWindow.java
package dev.reuss.tmdb.domain.trending.model;

/**
 * Time window for TMDB trending requests.
 */
public enum TrendingTimeWindow {
    DAY("day"),
    WEEK("week");

    private final String value;

    TrendingTimeWindow(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/trending/model/TrendingTimeWindow.kt
package dev.reuss.tmdb.domain.trending.model

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Time window for TMDB trending requests.
 */
enum class TrendingTimeWindow(
    @all:JsonProperty("value")
    val value: String,
) {
    DAY("day"),
    WEEK("week"),
}
